cmd/secret.go

```diff
@@ -0,0 +1,234 @@
+		registerSecretReq := &pb.RegisterSecretRequest{
+			ProjectName:   projectName,
+			SecretName:    secretName,
+			Value:         secretValue,
+			NamespaceName: namespaceName,
+		}
+		err = registerSecret(l, conf, registerSecretReq)
+		if err != nil {
+			if strings.Contains(err.Error(), "Internal desc = secret already exist") {
+				proceedWithUpdate := "Yes"
+				if !skipConfirm {
+					if err := survey.AskOne(&survey.Select{
+						Message: "Secret already exists, proceed with update?",
+						Options: []string{"Yes", "No"},
+						Default: "No",
+					}, &proceedWithUpdate); err != nil {
+						return err
+					}
+				}
+				if proceedWithUpdate == "Yes" {
+					updateSecretRequest := &pb.UpdateSecretRequest{
+						ProjectName:   projectName,
+						SecretName:    secretName,
+						Value:         secretValue,
+						NamespaceName: namespaceName,
+					}
+					return updateSecret(l, conf, updateSecretRequest)
+				} else {
+					l.Info(coloredNotice("Aborting..."))
+					return nil
+				}
+			}
+		}
+		return nil
+	}
+	return secretCmd
+}
+
+func getSecretName(args []string) (string, error) {
+	if len(args) < 1 {
+		return "", errors.New("secret name is required")
+	}
+	if strings.HasPrefix(args[0], models.SecretTypeSystemDefinedPrefix) {
+		return "", errors.New(fmt.Sprintf("secret name cannot be started with %s", models.SecretTypeSystemDefinedPrefix))
+	}
+	return args[0], nil
+}
+
+func getSecretValue(args []string, filePath string, encoded bool) (string, error) {
+	var secretValue string
+	if filePath == "" {
+		if len(args) < 2 {
+			return "", errors.New("secret value is required")
+		}
+		secretValue = args[1]
+	} else {
+		secretValueBytes, err := ioutil.ReadFile(filePath)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed when reading secret file %s", filePath)
+		}
+		secretValue = string(secretValueBytes)
+	}
+
+	if !encoded {
+		return base64.StdEncoding.EncodeToString([]byte(secretValue)), nil
+	} else {
+		if err := validateProperlyEncoded(secretValue); err != nil {
+			return "", err
+		}
+		return secretValue, nil
+	}
+}
+
+func validateProperlyEncoded(secretValue string) error {
+	if _, err := base64.StdEncoding.DecodeString(secretValue); err != nil {
+		return errors.New("value is not encoded. please use --base64 to let Optimus encode the secret for you.")
+	}
+	return nil
+}
+
+func registerSecret(l log.Logger, conf config.Provider, req *pb.RegisterSecretRequest) (err error) {
+	dialTimeoutCtx, dialCancel := context.WithTimeout(context.Background(), OptimusDialTimeout)
+	defer dialCancel()
+
+	var conn *grpc.ClientConn
+	if conn, err = createConnection(dialTimeoutCtx, conf.GetHost()); err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			l.Error(ErrServerNotReachable(conf.GetHost()).Error())
+		}
+		return err
+	}
+	defer conn.Close()
+
+	secretRequestTimeout, secretRequestCancel := context.WithTimeout(context.Background(), secretTimeout)
+	defer secretRequestCancel()
+
+	l.Info("please wait...")
```

Comment:
Should we use a spinner here? and in Update call?

Reply:
Using spinner